llvm/lib/SYCLLowerIR/PrepareSYCLNativeCPU.cpp

```diff
@@ -276,107 +283,31 @@ PreservedAnalyses PrepareSYCLNativeCPUPass::run(Module &M,
-    auto *Glob = M.getNamedGlobal(Entry.first);
+    auto *Glob = M.getFunction(Entry.first);
     if (!Glob)
       continue;
-    auto *ReplaceFunc = getReplaceFunc(M, StatePtrType, Entry.second);
+    auto *ReplaceFunc = getReplaceFunc(M, StatePtrType, Entry.second.first);
+    SmallVector<Instruction *> ToRemove;
     for (auto &Use : Glob->uses()) {
-      auto Funcs = getFunctionsFromUse(Use);
-      // Here we check that the use comes from a kernel function
-      // Todo: remove this check once this pass supports non-optimized modules
-      for (auto &Func : Funcs) {
-        if (!(Func->getCallingConv() == CallingConv::SPIR_KERNEL))
-          report_fatal_error("SYCL Native CPU currently supports only "
-                             "optimized modules, please enable optimizations "
-                             "and eventually increase the inlining threshold",
-                             false);
-      }
-      if (Funcs.empty()) {
-        // todo: use without a parent function?
-        continue;
-      }
-      for (auto &F : Funcs) {
-        auto NewCallIt = BuiltinCallMap.find(F);
-        CallInst *NewCall;
-        // check if we already inserted a call to our function
-        if (NewCallIt != BuiltinCallMap.end()) {
-          NewCall = NewCallIt->second;
-        } else {
-          auto *StateArg = getStateArg(F);
-          NewCall = llvm::CallInst::Create(
-              ReplaceFunc->getFunctionType(), ReplaceFunc, {StateArg},
-              "ncpu_builtin", F->getEntryBlock().getFirstNonPHI());
-          BuiltinCallMap.insert({F, NewCall});
-        }
-        User *Usr = Use.getUser();
-        if (auto *GEPOp = dyn_cast<GEPOperator>(Usr)) {
-          GEPOps.insert(GEPOp);
-        } else {
-          // Find the index of the builtin in the user's operand list
-          // We are guaranteed to find it since we are already iterating over
-          // the builtin's uses.
-          bool Found = false;
-          unsigned Index = 0;
-          for (unsigned I = 0; I < Usr->getNumOperands() && !Found; I++) {
-            if (Usr->getOperand(I) == Glob) {
-              Found = true;
-              Index = I;
-            }
-          }
-          assert(Found && "Unable to find builtin in operand list");
-          ToReplace.insert({Usr, {Index, NewCall}});
-        }
-      }
-    }
-
-    // Handle the non-constant builtin uses, simply replace the builtin with the
-    // return value of our function call
-    for (auto &Entry : ToReplace) {
-      unsigned Index = Entry.second.first;
-      CallInst *NewCall = Entry.second.second;
-      User *Usr = Entry.first;
-      Usr->setOperand(Index, NewCall);
+      auto I = dyn_cast<CallInst>(Use.getUser());
+      if (!I)
+        report_fatal_error("Unsupported Value in SYCL Native CPU\n");
+      if (I->getFunction()->getCallingConv() != llvm::CallingConv::SPIR_KERNEL)
+        report_fatal_error(
+            "SYCL Native CPU currently doesn't support unlined "
```

Comment:
Does `unlined` mean `non-inlined`? I'm sorry if this is a silly question, but I never heard this term before.